test/unit/org/apache/cassandra/cql3/validation/entities/SecondaryIndexTest.java

```diff
@@ -1560,4 +1613,77 @@ public IndexBlockingOnInitialization(ColumnFamilyStore baseCfs, IndexMetadata in
             return super.getInvalidateTask();
         }
     }
+    
+    /**
+     * <code>StubIndex</code> that blocks during the initialization.
+     */
+    public static class BlockingStubIndex extends StubIndex
+    {
+        private final CountDownLatch latch = new CountDownLatch(1);
+
+        public BlockingStubIndex(ColumnFamilyStore baseCfs, IndexMetadata indexDef)
+        {
+            super(baseCfs, indexDef);
+        }
+
+        @Override
+        public Callable<?> getInitializationTask()
+        {
+            return () -> {
+                latch.await();
+                return null;
+            };
+        }
+
+        @Override
+        public Callable<?> getInvalidateTask()
+        {
+            latch.countDown();
+            return super.getInvalidateTask();
+        }
+    }
+
+    /**
+     * <code>CassandraIndex</code> that only supports reads. Could be intentional or a result of a bad init
+     */
+    public static class ReadOnlyIndex extends StubIndex
+    {
+        public ReadOnlyIndex(ColumnFamilyStore baseCfs, IndexMetadata indexDef)
+        {
+            super(baseCfs, indexDef);
+        }
+
+        @Override
+        public Callable<?> getInitializationTask()
+        {
+            return null;
+        }
+
+        public boolean supportsLoad(LoadType load)
+        {
+            return load.equals(LoadType.READ);
+        }
+    }
+
+    /**
+     * <code>CassandraIndex</code> that only supports writes. Could be intentional or a result of a bad init
```

Comment:
Nit: is not a `CassandraIndex`:
```suggestion
     * {@code StubIndex} that only supports writes. Could be intentional or a result of a bad init.
```